### DISCUSSION TITLE
fail to convert the ROS type in a YARP compatible type

### DISCUSSION THREAD

**Discussion by jimingda** [robotology/community#676] [Q&A] [answered]
Hi, 

After following the yarp-ros [tutorial](https://www.yarp.it/latest/rostypes_tutorial_portable.html), I defined a `SharedData.msg` file in my working directory, but encountered an error when using the `yarpidl_rosmsg SharedData.msg` command.

Here is the output of the command:
```
icub@icubsrv:~/workspace/rostest$ yarpidl_rosmsg SharedData.msg
Failed to open yarp/rosmsg/SharedData.h for writing
icub@icubsrv:~/workspace/rostest$ which yarpidl_rosmsg
/usr/local/src/robot/robotology-superbuild/build/install/bin/yarpidl_rosmsg
icub@icubsrv:~/workspace/rostest$ which rosidl
```
Could you please advise on any points I should pay attention to in order to resolve this issue?

Thank you in advance.

**Comment by jimingda** [comment]
As for [yarp_with_ros_configure](https://www.yarp.it/latest/yarp_with_ros_configure.html), how/where to enable these parameters ?

**Comment by Nicogene** [comment]
Hi @jimingda,

which YARP version are you using?
Not that the ros part of yarp has been moved:
- https://github.com/robotology/yarp/pull/3005

@elandini84 and @randaz81 maybe knows the source of your issue

**Reply by jimingda** [reply]
My yarp version is `YARP version 3.7.0`

**Reply by jimingda** [reply, marked as answer]
I think the problem has been resolved since i can finished the ros yarp message test on the tutorial.
